### PR TITLE
These are my code review suggestions

### PR DESCRIPTION
**Pull request by phanga**
Description/Analysis of the Project
This is an app that identifies different recycling places and can add them to favorites.

Was the program available in UC Github on time?
Yes

Is the program documented/commented well enough for you to understand?
Yes

Does the program compile?

Added curly braces were needed to be added for the main activity to be built.

Rationale Behind Changes

I had done a comparison to my own project and saw that there was some pieces missing for Koin in the project such as the Application() class. Also added was a JVM field for the view model which also dealt with some missing pieces for Koin. Finally as I was getting the project to build correctly I spotted a typo and fixed it according to best practices.

3 Commits

Commit #1
https://github.com/phanga/MyPerishablePlannerV21001/commit/66c1a98a2035e5ff6e3feff834589cb2a51db18e
Commit #2
https://github.com/phanga/MyPerishablePlannerV21001/commit/a722fcde87a826abf4022259c9f743fe453293b0
Commit #3 
https://github.com/phanga/MyPerishablePlannerV21001/commit/86367c6ee9c4be00d2e29a3c543b0342880418c9

Things I Learned

Making additions to previous Koin practices

Learning about Application constructors in Kotlin
Practicing and modifying best practices

### CODE REVIEW

File: app/src/main/java/com/findmyrecycling/AppModule.kt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+import com.findmyrecycling.service.ProductService
+import org.koin.androidx.viewmodel.dsl.viewModel
+import org.koin.dsl.module
+
+@JvmField
+val appModule = module {
+    viewModel { MainViewModel(get()) }
+    single <ProductService>{ ProductService() }
+}
```
Comment (phanga) [inline app/src/main/java/com/findmyrecycling/AppModule.kt:11]: Added AppModule.kt file for Koin based on these references: https://kotlinlang.org/docs/java-to-kotlin-interop.html#instance-fields
https://blog.mindorks.com/kotlin-koin-tutorial

File: app/src/main/java/com/findmyrecycling/FindMyRecyclingApplication.kt [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            super.onCreate()
+            GlobalContext.startKoin{
+                androidLogger(if (BuildConfig.DEBUG) Level.ERROR else Level.NONE)
+                androidContext(this@FindMyRecyclingApplication)
+                modules(appModule)
+            }
+
+        }
+    }
+}
```
Comment (phanga) [inline app/src/main/java/com/findmyrecycling/FindMyRecyclingApplication.kt:22]: Added FindMyRecylingApplication.kt file to link with the AppModule.kt file, this again is a file needed for Koin, reference for Application constructors was used from the following site: 
https://developer.android.com/reference/kotlin/android/app/Application

File: app/src/main/java/com/findmyrecycling/MainActivity.kt [hunk 3]
```diff
@@ -29,84 +29,92 @@ class MainActivity : ComponentActivity() {
...
+        fun signIn() {
+            val providers = arrayListOf(AuthUI.IdpConfig.EmailBuilder().build())
+            val signInIntent = AuthUI.getInstance()
+                .createSignInIntentBuilder()
+                .setAvailableProviders(providers)
+                .build()
+
+            signInLauncher.launch(signInIntent)
+        }
+
```
Comment (phanga) [inline app/src/main/java/com/findmyrecycling/MainActivity.kt:106]: From line 97- 105 is where the edit for this file change is, there was a typo on line 99 with the signInIntent which aslo gets called on line 104, reference for this code change comes from best practices found here: 
https://kotlinlang.org/docs/coding-conventions.html

File: app/src/main/java/com/findmyrecycling/MainActivity.kt [hunk 4]
```diff
@@ -29,84 +29,92 @@ class MainActivity : ComponentActivity() {
...
+        Column {
+            OutlinedTextField(
+                value = recyclable,
+                onValueChange = { recyclable = it },
+                label = { Text(stringResource(R.string.recyclable)) },
+                modifier = Modifier
+                    .fillMaxWidth()
+                    .padding(start = 4.dp, end = 4.dp, top = 2.dp, bottom = 2.dp)
+                    .border(
+                        width = 1.dp,
```
Comment (discospiff) [inline app/src/main/java/com/findmyrecycling/MainActivity.kt:57]: What's the nature/rationale for this change?

**Comment by discospiff** [comment]
From this review:
- Koin is a good idea to implement fully... though I'm not certain it has been brought all the way through to the Activity in this branch.  If not, I encourage the team to finish it.
- I'm not entirely clear on the large change in MainActivity... would like the reviewer to give a bit more detail here.

File: app/src/main/java/com/findmyrecycling/MainActivity.kt (hunk 4 above)
Comment (phanga) [reply app/src/main/java/com/findmyrecycling/MainActivity.kt:57]: There was no big change but there had been bracket misplacement in the code which kept the MainActivity from being built. So there may be a big change as seen but in reality it is minor just getting the curly braces in the right places for the project to compile.

**Pull request closed without merging**